clusterhost.go

```diff
@@ -28,13 +32,41 @@ func NewClusterHost(
 
 	connman := connmgr.NewConnManager(cfg.ConnMgr.LowWater, cfg.ConnMgr.HighWater, cfg.ConnMgr.GracePeriod)
 
+	relayOpts := []relay.RelayOpt{relay.OptDiscovery}
+	if cfg.EnableRelayHop {
+		relayOpts = append(relayOpts, relay.OptHop)
+	}
+
+	opts := []libp2p.Option{
+		libp2p.ListenAddrs(cfg.ListenAddr...),
+		libp2p.NATPortMap(),
+		libp2p.ConnectionManager(connman),
+		libp2p.Routing(func(h host.Host) (routing.PeerRouting, error) {
+			return dht.New(ctx, h)
+		}),
+		libp2p.EnableRelay(relayOpts...),
+		libp2p.EnableAutoRelay(),
+	}
+
+	for _, addr := range cfg.ListenAddr {
+		out := false
+		for _, protocol := range addr.Protocols() {
+			if protocol.Code == ma.P_QUIC {
+				opts = append(opts, libp2p.Transport(libp2pquic.NewTransport))
```

Comment:
I think you are replacing all transports with QUIC. Use `ChainOptions` to prepend the quick transport to the default transports.